File: src/main/scala/com/socrata/controller/VersionController.java
package com.socrata.controller;

import com.socrata.api.VersionApi;
import com.socrata.models.VersionResponse;
import io.smallrye.mutiny.Uni;
import jakarta.enterprise.context.ApplicationScoped;

@ApplicationScoped
public class VersionController implements VersionApi {
    @Override
    public Uni<VersionResponse> getVersion() {
       return Uni.createFrom().nullItem();
    }
}

File: src/main/scala/com/socrata/controller/VersionController.scala
package com.socrata.controller

import com.socrata.api.{VersionApi, VersionResponse}
import com.socrata.config.VersionConfig
import io.smallrye.mutiny.Uni
import jakarta.enterprise.context.ApplicationScoped

@ApplicationScoped
class VersionController(config: VersionConfig) extends VersionApi {
  override def getVersion(): Uni[VersionResponse] = {
    //Simply return our config for now
    Uni.createFrom().item(
      config
    )
  }
}
